src/psyclone/psyGen.py

```diff
@@ -5863,6 +5952,29 @@ def __str__(self):
             result += str(entity)
         return result
 
+    def reference_accesses(self, var_accesses):
+        '''Get all variable access information from this node. The assigned-to
+        variable will be set to 'WRITE'.
+        :param var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+        '''
+
+        # It is important that a new instance is used to handle the LHS,
+        # since an assert in 'change_read_to_write' makes sure that there
+        # is only one access to the variable!
+        accesses_left = VariablesAccessInfo(var_accesses.get_location())
+        self.lhs.reference_accesses(accesses_left)
+
+        # Now change the (one) access to the assigned variable to be WRITE:
+        var_info = accesses_left.get_varinfo(self.lhs.name)
+        var_info.change_read_to_write()
+
+        # Merge the data (that shows now WRITE for the variable) with the
+        # parameter to this function:
+        var_accesses.merge(accesses_left)
+        self.rhs.reference_accesses(var_accesses)
```

Comment:
Does this work OK for e.g. `my_var = my_var + 1`? i.e. `my_var` is read before it is written. Would the `location` for the read and write be the same in this situation?

Reply:
Yes, it works, and yes, same location, and I have a test case for that (and actually a minor fix to make sure that the read is coming before the write).